hexrd/instrument.py

```diff
@@ -725,8 +729,11 @@ def __init__(self, instrument_config=None,
                             raise RuntimeError(
                                 "problem with distortion specification"
                             )
+                DetectorClass = PlanarDetector
+                if detector_type.lower() == 'cylindrical':
+                     DetectorClass = CylindricalDetector
```

Comment:
We might want to put a dict of detector types at the top of the file like:
```
DETECTOR_TYPES = {
    'planar': PlanarDetector,
    'cylindrical': CylindricalDetector,
}
```

And then this code here would have some error checking as well:

```python
    if detector_type.lower() not in DETECTOR_TYPES:
        raise NotImplementedError(f'Unknown detector type: {detector_type}')

    DetectorClass = DETECTOR_TYPES[detector_type.lower()]
```

That error checking can be helpful because if a user mistypes the word `cylindrical`, they would probably prefer to see an error rather than it continuing as normal with a `PlanarDetector` (when they might think it is using a `CylindricalDetector`).

Reply:
Implemented these changes.